### PR TITLE
[BUGFIX] Avoid NullPointer exception

### PR DESCRIPTION
**Pull request by linawolf**
Calling an unkown directive with field options causes a Nullpointer exception. We cannot rely on it to be set here. Unknown directives already cause warning

### CODE REVIEW

File: lib/Parser/DocumentParser.php [hunk 1]
```diff
@@ -505,7 +505,7 @@ private function parseLine(string $line): bool
                 }
 
                 if ($this->lineChecker->isFieldOption($line)) {
-                    if ($this->fieldOption !== null) {
+                    if ($this->directive !== null && $this->fieldOption !== null) {
```
Comment (greg0ire) [inline lib/Parser/DocumentParser.php:508]: So we silently don't set the option? Should maybe an error end up in the error manager or something?

Comment (linawolf) [reply lib/Parser/DocumentParser.php:508]: Non-existent directives already throw warnings. I don't see a reason, why a directive that has field options should throw a stronger errror then a directive without. Non-existent directive is non-existent directive in my eyes?

Comment (greg0ire) [reply lib/Parser/DocumentParser.php:508]: I wasn't aware warnings were already thrown. In that case, this should be fine 👍

**Review by greg0ire — APPROVED** [review]

**Comment by SenseException** [comment]
Please cover this case with a test. :test_tube:

**Comment by linawolf** [comment]
@SenseException here you go

**Review by SenseException — CHANGES_REQUESTED** [review]
The test files don't make phpunit fail when I revert to the old code. The tests are supposed to fail on `if ($this->fieldOption !== null) {` and work on `if ($this->directive !== null && $this->fieldOption !== null) {`

**Comment by linawolf** [comment]
@SenseException, Ok, should be fixed now. The whole thing is a little tricky as weather or not the Nullpointer exception is reached depends on the settings of the error manager: If the error manager throws an Exception on finding the error of the non-existent directive then we never get to the code where the document parser tries to set the fields on the non-existent directive.

So this error only happens when you disable throwing exceptions on finding errors.

**Review by SenseException — APPROVED** [review]
Well done :+1:

**Comment by SenseException** [comment]
Once again, thank you @linawolf

**Pull request merged**